Library/ViewModels/CommentsViewModel.swift

```diff
@@ -105,99 +99,144 @@ public final class CommentsViewModel: CommentsViewModelType,
     self.commentComposerViewHidden = currentUser.signal
       .map { user in user.isNil }
 
-    let postCommentEvent = initialProject
-      .takePairWhen(self.postCommentButtonTappedProperty.signal.skipNil())
-      .switchMap { project, comment in
-        AppEnvironment.current.apiService
-          .postComment(input: .init(
-            body: comment,
-            commentableId: project.graphID
-          ))
-          .materialize()
-      }
-
-    // TODO: Handle error and success states appropriately for the datasource item
-    self.postCommentSuccessful = postCommentEvent.values().map { $0 }
-    self.errorMessage = postCommentEvent.errors().map { $0.errorMessages.first }.skipNil()
-
     let isCloseToBottom = self.willDisplayRowProperty.signal.skipNil()
-      .map { row, total in row >= total - 3 }
+      .map { row, total -> Bool in
+        // Prevent paging when only the empty state cell is shown.
+        guard total > 1 else { return false }
```

Comment:
Hmm, yeah maybe - I might just add a `TODO` here for this to be improved when we do the pagination error handling because I think that'll touch this part.